debezium-core/src/main/java/io/debezium/transforms/extractnewstate/DefaultDeleteHandlingStrategy.java

```diff
@@ -57,6 +58,9 @@ public R handleDeleteRecord(R record) {
                 // a record only with null value that by JDBC connector is treated as a flattened delete.
                 // Any change to this behavior can have impact on JDBC connector.
                 return afterDelegate.apply(record);
+            case REWRITE_DELETES:
+                LOGGER.trace("Delete message {} requested to be converted to tombstone", maybeRedactSensitiveData(record.key()));
+                return tombstoneDelegate.apply(record);
```

Comment:
@vaibhav-yb There is no need for the delegate, you can just inline `record.newRecord` here.